src/txn/txn.c

```diff
@@ -1144,7 +1148,10 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
      * another transaction when the transaction tries to resolve the subsequent operations on the
      * same key.
      */
-    for (; upd != NULL && upd->txnid != txn->id; upd = upd->next)
+    for (; upd != NULL &&
+         (upd->txnid != txn->id &&
```

Comment:
Can we just do upd->txnid != WT_TXN_ABORTED?

Reply:
```
Hence the update chain could contain the prepared updates by another transaction when the transaction tries to resolve the subsequent operations on the same key.
```

This should not happen as we now only resolve the same key once.

Reply:
done